controllers/external/util.go

```diff
@@ -114,6 +115,61 @@ func CloneTemplate(ctx context.Context, in *CloneTemplateInput) (*corev1.ObjectR
 	return GetObjectReference(to), nil
 }
 
+// CreateFromTemplateInput is the input to CloneTemplate.
+type CreateFromTemplateInput struct {
```

Comment:
the external package is called "external" in the sense that it manipulates references to externals resources. I wonder about the value of having this as a public library vs working towards making it an internal library.

Reply:
Good point. I'm not sure if all functions are just used internally though.

But it looks like at least CloneTemplate can be made internal instead: https://cs.k8s.io/?q=external.CloneTemplate&i=nope&files=&excludeFiles=&repos=

Reply:
Looks like the talos provider is using this function: https://grep.app/search?q=external.CloneTemplate